### PR TITLE
Fixed the spinner for `Unlock User Keyring`

### PR DESCRIPTION
**Pull request by 4rjunc**
## :mag: Overview

This PR is a fix for the issue #200 . 

## :bulb: Proposed Changes

*Detail the proposed changes, including new features, bug fixes, or improvements. Explain how these changes impact the project, including any internal structure alterations or refactorings.*

## :framed_picture: Screenshots or Demo

Before : the `Unlock button will stay still for few seconds` like the video shown in issue #200 

Fix 

https://github.com/user-attachments/assets/6385906c-676d-40ae-991f-ffe378e4540d

## :memo: Release Notes

*Summarize the changes in a user-friendly manner. Highlight new features, bug fixes, and any breaking changes, including migration steps or deprecated functionalities.*

## :question: Open Questions

*If there are aspects of the changes that you're unsure about or would like feedback on, list them here.*

### :test_tube: Testing

*Describe the testing strategy. List new tests added, existing tests modified, and any testing gaps.*

### :dart: Reviewer Focus

*Guide the reviewer on where to start the review process. Suggest specific files, modules, or functionalities to focus on as entry points.*

### :heavy_plus_sign: Additional Context

*Provide any additional information that might be helpful for reviewers and future contributors, such as links to related issues, discussions, or resources.*

## :sparkles: How to Test the Changes Locally

When the `Unlock User Keyring` Box appears. Enter the sudo password and you will see the spinner instead unlock 🔓 icon 

### :green_heart: Did You...

- [ ✅ ] Ensure linting passes (code style checks)?
- [ ] Update dependencies and lockfiles (if required)
- [ ] Regenerate graphql schema and types (if required)
- [ ] Verify the app builds locally?
- [✅ ] Manually test the changes on different browsers/devices? 
   - Checked in Chrome and Brave

### CODE REVIEW

File: frontend/components/auth/UnlockKeyringDialog.tsx [hunk 1]
```diff
@@ -46,23 +46,24 @@ export default function UnlockKeyringDialog(props: { organisation: OrganisationT
...
+          setUnlocking(false)
+          reset()
+          closeModal()
+          resolve(true) // Resolve the promise successfully
+        } catch (e) {
+          console.error(e)
+          setUnlocking(false)
+          reject(e) // Reject the promise with the error
+        }  
+      }, 2000);
```
Comment (rohan-chaturvedi) [inline frontend/components/auth/UnlockKeyringDialog.tsx]: @4rjunc Is it possible to use a smaller delay? 2 seconds seems like a lot. How about ~300ms?

Comment (aarjn) [reply frontend/components/auth/UnlockKeyringDialog.tsx]: Okay will change it.

Comment (aarjn) [reply frontend/components/auth/UnlockKeyringDialog.tsx]: Pushed the changes.

Comment (rohan-chaturvedi) [reply frontend/components/auth/UnlockKeyringDialog.tsx]: @4rjunc One side effect of this change is that even when the device is trusted, this delay causes the popup to show for much longer than before: 

[Screencast from 07-27-2024 10:26:04 PM.webm](https://github.com/user-attachments/assets/38ec6302-8ac7-4dfc-9c96-91dca49f49cb)


I'd like to try and reduce the timeout delay as much as possible. I've tested `100`ms delay locally and it seems to work fine. Could you try that on your system as well and see if it works correctly? If that works, then lets go ahead with that.

Comment (aarjn) [reply frontend/components/auth/UnlockKeyringDialog.tsx]: Screenshot 2024-07-27 at 23 48 47
When the device is trusted , `trustDevice` variable is false. I got confused. I have checked with 100ms it works fine.

Comment (rohan-chaturvedi) [reply frontend/components/auth/UnlockKeyringDialog.tsx]: Great! Could you push that change? We can get this merged.

Comment (aarjn) [reply frontend/components/auth/UnlockKeyringDialog.tsx]: I couldn't get the fix for the time lag issue when the device is trusted. I have reduced the timeout from 300 -> 100

**Review by rohan-chaturvedi — APPROVED** [review]
Great work @4rjunc !

**Comment by 4rjunc** [comment]
@rohan-chaturvedi , will this be considered for FOSS Hack 2024? and can you please check the DM in Slack?

**Pull request merged**